src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(5);
+
+        switch (randomNumber) {
+            case 0:
+                return new Circle(random.nextInt(),
+                        colorSupplier.getRandomColor());
+            case 1:
+                return new IsoscelesTrapezoid(random.nextInt(),
+                        random.nextInt(),
+                        random.nextInt(),
+                        colorSupplier.getRandomColor());
+            case 2:
+                return new Rectangle(random.nextInt(),
+                        random.nextInt(),
+                        colorSupplier.getRandomColor());
+            case 3:
+                return new RightTriangle(random.nextInt(),
+                        random.nextInt(),
+                        colorSupplier.getRandomColor());
+            default:
+                return new Square(random.nextInt(),
+                        colorSupplier.getRandomColor());
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10,"white");
```

Comment:
Use constants for magic numbers. The value '10' and the string "white" should be defined as constants.